library/psa_crypto.c

```diff
@@ -5177,32 +5195,48 @@ static psa_status_t psa_hkdf_input( psa_hkdf_key_derivation_t *hkdf,
             }
         case PSA_KEY_DERIVATION_INPUT_SECRET:
             /* If no salt was provided, use an empty salt. */
-            if( hkdf->state == HKDF_STATE_INIT )
+            if( PSA_ALG_IS_HKDF_EXPAND( kdf_alg ) )
```

Comment:
I think the comment above is misplaced now, and should go in the `else` branch, above `if( hkdf->state == HKDF_STATE_INIT )`.